### PR TITLE
[EventHub] update get_buffered_event_count docstring

### PR DESCRIPTION
**Pull request by swathipil**

### CODE REVIEW

File: sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py [hunk 1]
```diff
@@ -831,8 +831,11 @@ def get_buffered_event_count(self, partition_id: str) -> Optional[int]:
         """
         The number of events that are buffered and waiting to be published for a given partition.
          Returns None in non-buffered mode. **NOTE: This method should only be used for debugging
-         purposes. If an invalid `partition_id` is passed in, then a value of 0 will be returned
-         which may be misleading.**
+         purposes. Returns the approximate number of events in the buffer for a partition ID,
+         because it is reporting the state of a background thread.  For a partition ID that has no
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py:835]: Minor nitpick - I would probably rephrase these first two sentences:
"The event buffer is processed in a background thread, therefore the number of events in the buffer reported by this API should be considered only an approximation and is only recommend for use in debugging."

File: sdk/eventhub/azure-eventhub/azure/eventhub/aio/_producer_client_async.py [hunk 2]
```diff
@@ -828,8 +828,11 @@ def get_buffered_event_count(self, partition_id: str) -> Optional[int]:
         """
         The number of events that are buffered and waiting to be published for a given partition.
          Returns None in non-buffered mode. **NOTE: This method should only be used for debugging
-         purposes. If an invalid `partition_id` is passed in, then a value of 0 will be returned
-         which may be misleading.**
+         purposes. Returns the approximate number of events in the buffer for a partition ID,
+         because it is reporting the state of a background thread.  For a partition ID that has no
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub/azure/eventhub/aio/_producer_client_async.py:832]: Background coroutine? Or did we use threads for async as well?

Comment (swathipil) [reply sdk/eventhub/azure-eventhub/azure/eventhub/aio/_producer_client_async.py:832]: I think it should be background coroutine @kashifkhan?

**Review by annatisch — APPROVED** [review]
Thanks :)

**Pull request merged**